Dockerfile

```diff
@@ -0,0 +1,67 @@
+# This is a multi-stage build which requires Docker 17.05 or higher
+FROM python:3.7-alpine as molecule-builder
+
+WORKDIR /usr/src/molecule
+
+ENV PACKAGES="\
+    gcc \
+    git \
+    libffi-dev \
+    make \
+    musl-dev \
+    openssl-dev \
+    "
+RUN \
+    apk update && \
+    apk add --update --no-cache ${PACKAGES}
+
+ADD . .
+RUN \
+    pip wheel \
+    -w dist \
+    .[azure,docker,docs,ec2,gce,lxc,openstack,vagrant,windows]
+
+# ✄---------------------------------------------------------------------
+# This is an actual target container:
+
+FROM python:3.7-alpine
+LABEL maintainer "Ansible <info@ansible.com>"
+
+ENV PACKAGES="\
+    docker \
+    ruby \
+    "
+ENV BUILD_DEPS="\
+    gcc \
+    libc-dev \
+    make \
+    ruby-dev \
+    ruby-rdoc \
+    "
+
+ENV PIP_INSTALL_ARGS="\
+    --only-binary :all: \
+    --no-index \
+    -f /usr/src/molecule/dist \
+    "
+ENV PIP_PACKAGES="\
+    molecule[azure,docker,docs,ec2,gce,lxc,openstack,vagrant,windows] \
+    "
+
+ENV GEM_PACKAGES="\
+    rubocop \
+    "
+
+COPY --from=molecule-builder \
+    /usr/src/molecule/dist \
+    /usr/src/molecule/dist
+
+RUN \
+    apk update \
+    && apk add --update --no-cache ${BUILD_DEPS} ${PACKAGES} \
+    && pip install ${PIP_INSTALL_ARGS} ${PIP_PACKAGES} \
+    && gem install ${GEM_PACKAGES} \
+    && apk del --no-cache ${BUILD_DEPS} \
+    && rm -rf /var/cache/apk/* /root/.cache
+
```

Comment:
The original Dockerfile had `USER molecule`

Reply:
I know but that's totally unnecessary and sometimes breaks assumptions.

Reply:
Is it possible with this change we could break existing users?

Reply:
Unlikely

Reply:
Let's remember to add this to the change log entry.

Reply:
This makes it harder to test `delegate_to: localhost` tasks in roles if the whole container runs as root.

Reply:
Why?